### PR TITLE
Fedora 35 image

### PR DESCRIPTION
**Pull request by mvollmer**
* [x] image-refresh fedora-35

### CODE REVIEW

**Comment by mvollmer** [comment]
This fails right now since https://download.fedoraproject.org/pub/fedora/linux/releases/35/Cloud/x86_64/images/ doesn't exist yet.  But this is an easy way to try once in a while.

**Comment by marusak** [comment]
You may add f-35 into testmap in the meantime

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-03.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210812-134242/

File: images/scripts/fedora-35.bootstrap [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301 USA.
+
+set -eux
+
+URL='https://download.fedoraproject.org/pub/fedora/linux/releases/35/Cloud/x86_64/images/'
```
Comment (marusak) [inline images/scripts/fedora-35.bootstrap:22]: For the record, this url won't exist for some time, first it will live under https://download.fedoraproject.org/pub/fedora/linux/development/35/Cloud/x86_64/images/

Comment (marusak) [reply images/scripts/fedora-35.bootstrap:22]: ha, and there are even composes already!

Comment (mvollmer) [reply images/scripts/fedora-35.bootstrap:22]: Ah, I see.

**Comment by cockpituous** [comment]
image-refresh fedora-34 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-34-20210812-140101

**Comment by cockpituous** [comment]
image-refresh in progress on 3-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210812-140202/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-2303-20210812-140202/

File: lib/testmap.py [hunk 2]
```diff
@@ -58,6 +58,7 @@
             'fedora-testing/dnf-copr',
             'rhel-9-0-distropkg',
             'fedora-34/firefox-devel',
+            'fedora-35',
```
Comment (marusak) [inline lib/testmap.py:61]: This needs to be separate PR as when you want to trigger `fedora-35` context in this PR, bots already need to know about it.

**Comment by mvollmer** [comment]
This went pretty far, but couldn't synchronize selinux policies between OS and mock.

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-01.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210816-081818/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-2303-20210816-081818/

**Comment by mvollmer** [comment]
Something wrong with the signatures.

**Comment by cockpituous** [comment]
image-refresh in progress on 4-ci-srv-04.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210816-091919/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-2303-20210816-091919/

**Comment by cockpituous** [comment]
image-refresh in progress on 1-ci-srv-06.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210818-070909/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-2303-20210818-070909/

**Comment by cockpituous** [comment]
image-refresh in progress on 2-ci-srv-02.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210818-073434/

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-2303-20210818-073434/

**Comment by mvollmer** [comment]
`mock` is still building for Fedora rawhide, but has only keys for Fedora 35.  There are two symlinks: 

    default.cfg -> fedora-35-x86_64.cfg (seems correct)
    fedora-35-x86_64.cfg -> fedora-rawhide-x86_64.cfg (seems wrong)

I'll ask around what's going on here.

**Comment by mvollmer** [comment]
> I'll ask around what's going on here.

https://github.com/rpm-software-management/mock/issues/762

File: images/scripts/fedora.setup [hunk 3]
```diff
@@ -155,11 +155,13 @@ echo "config_opts['use_bootstrap'] = False" >>/etc/mock/site-defaults.cfg
...
-our_selinux_policy_version="$(rpm -q --qf '%{EVR}' selinux-policy)"
-selinux_mock_packages="selinux-policy{,-devel,-targeted}-${our_selinux_policy_version}"
-su builder -c "/usr/bin/mock --addrepo='$archive_repo' --install ${selinux_mock_packages}"
+if [ "${IMAGE}" != "fedora-35" ]; then
+    # we need to ensure that mock's selinux-policy is the same as the host
+    archive_repo="https://fedoraproject-updates-archive.fedoraproject.org/fedora/$version/x86_64/"
+    our_selinux_policy_version="$(rpm -q --qf '%{EVR}' selinux-policy)"
+    selinux_mock_packages="selinux-policy{,-devel,-targeted}-${our_selinux_policy_version}"
+    su builder -c "/usr/bin/mock --addrepo='$archive_repo' --install ${selinux_mock_packages}"
+fi
```
Comment (allisonkarlitskaya) [inline images/scripts/fedora.setup:164]: This is out of date since #2313

**Comment by mvollmer** [comment]
> > I'll ask around what's going on here.
> 
> [rpm-software-management/mock#762](https://github.com/rpm-software-management/mock/issues/762)

I have put a workaround in so that we can proceed with running some tests.

**Comment by cockpituous** [comment]
image-refresh in progress on 2-cockpit-8.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210818-090808/

File: images/scripts/fedora.setup (hunk 3 above)
Comment (mvollmer) [reply images/scripts/fedora.setup]: Fixed.

**Comment by cockpituous** [comment]
Task failed: https://logs.cockpit-project.org/logs/image-refresh-2303-20210818-090808/

**Comment by cockpituous** [comment]
image-refresh in progress on 3-cockpit-11.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210818-103434/

**Comment by cockpituous** [comment]
image-refresh fedora-35 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-35-20210818-105050

**Comment by mvollmer** [comment]
Quick notes about test failures:

TestPages.testBasic: text changed from "Fedora documentation" to "Fedora Linux documentation"
TestMultiMachine.testFIPS: new selinux message
TestStorageResize.testGrowShrinkEncryptedHelp, others: need to copy naughty patterns
TestTerminal.testBasic: rhsm.service has failed and obscures the prompt matching
TestUpdates.testKpatch: kpatch is not installed...?

**Comment by marusak** [comment]
> TestUpdates.testKpatch: kpatch is not installed...?

https://github.com/cockpit-project/cockpit/blob/main/test/verify/check-packagekit#L40

**Comment by mvollmer** [comment]
TestUpdatesSubscriptions: Broken by rhsm.service, https://bugzilla.redhat.com/show_bug.cgi?id=1995465 -> naughties

**Comment by mvollmer** [comment]
TestUpdates.testPackageKitCrash: allow "Found module" messages in backtraces.

**Comment by mvollmer** [comment]
TestHostSwitching.testBasic  -> copy naugthy from fedora-34

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-7.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210820-081919/

**Comment by mvollmer** [comment]
> > > I'll ask around what's going on here.
> > 
> > 
> > [rpm-software-management/mock#762](https://github.com/rpm-software-management/mock/issues/762)
> 
> I have put a workaround in so that we can proceed with running some tests.

This is no longer needed.

**Comment by cockpituous** [comment]
image-refresh fedora-35 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-35-20210820-083434

**Comment by cockpituous** [comment]
image-refresh in progress on 4-cockpit-10.
Log: https://logs.cockpit-project.org/logs/image-refresh-2303-20210820-094848/

**Comment by cockpituous** [comment]
image-refresh fedora-35 done: https://github.com/cockpit-project/bots/commits/image-refresh-fedora-35-20210820-100303

**Comment by mvollmer** [comment]
Tests in https://github.com/cockpit-project/cockpit/pull/16246

File: naughty/fedora-35/2323-rhsm-crashes-3 [hunk 4]
```diff
@@ -0,0 +1,3 @@
+  File "check-system-terminal", line *, in testBasic
+    wait_line(n, prompt)
+*actual text: [systemd]
```
Comment (marusak) [inline naughty/fedora-35/2323-rhsm-crashes-3:3]: This is still not the most robust one as if different systemd service starts to fail, it will reuse the same pattern.

Comment (mvollmer) [reply naughty/fedora-35/2323-rhsm-crashes-3:3]: Ok, convinced!  Let's modify wait_line to output some context and match on that.

**Comment by marusak** [comment]
Looks good! Running tests

**Comment by marusak** [comment]
https://logs.cockpit-project.org/logs/pull-2303-20210825-090226-7d999bdd-fedora-35-cockpit-project-cockpit/log.html#269

One unexpected message, seems like naughty #2141

**Comment by mvollmer** [comment]
> One unexpected message, seems like naughty #2141

Yeah, I have just added it to "allowed messages"... but that might be the wrong approach. Should we turn all the unexpected selinux messages into naughties, actually?

**Comment by marusak** [comment]
> Yeah, I have just added it to "allowed messages"... but that might be the wrong approach. Should we turn all the unexpected selinux messages into naughties, actually?

Not really. If the message happens too often, then it it better in allowed messages. If it happens now and then, then naughty is better.
Reasoning: Naughties are auto-maintained. Bots will close them and we don't need to track them. So they are superior. But on every occurance we comment on github issue, which creates traffic, thus if it would commend 50 times for every context, that would be just pointelssly expensive.

**Review by marusak — APPROVED** [review]
Thanks, assuming green F-35 context
